### PR TITLE
Updates to install documentation

### PR DESCRIPTION
**Pull request by dalcinl**

### CODE REVIEW

**Comment by dalcinl** [comment]
@YarShev Is this good enough for you? [[link]](https://mpi4py--348.org.readthedocs.build/en/348/install.html#using-conda)

File: docs/source/install.rst [hunk 1]
```diff
@@ -108,8 +108,31 @@ version the MPI implementation advertises to support (via the
...
+   ``"meson-python"`` to use the `meson-python`_ build backend.
+
+When using the `meson-python`_ build backend, mpi4py delegates build
+tasks to the `Meson`_ build system.
+
+.. warning::
+
+   mpi4py support for the `meson-python`_ build backend is
+   experimental. For the time being, users must set the :envvar:`CC`
+   environment variable to the command or path correspondin to the
```
Comment (YarShev) [inline docs/source/install.rst:131]:
```suggestion
   environment variable to the command or path corresponding to the
```

**Comment by YarShev** [comment]
> @YarShev Is this good enough for you? [[link]](https://mpi4py--348.org.readthedocs.build/en/348/install.html#using-conda)

@dalcinl, thanks for pinging me on this matter. I added some note in unidist too [[link](https://github.com/modin-project/unidist/pull/280/files)]. Do you think, maybe, it is worth adding a note why users should look into the External MPI libraries section of the conda-forge docs, for example, it is encouraged to use user/system provided MPI binaries to get ultimate performance?

**Comment by dalcinl** [comment]
@YarShev What about now? [[link]](https://mpi4py--348.org.readthedocs.build/en/348/install.html#using-conda)

**Review by leofang — APPROVED** [review]

**Pull request merged**